glass/shells.py

```diff
@@ -195,16 +196,31 @@ class RadialWindow(NamedTuple):
     zeff
         Effective redshift of the window.
 
-    Methods
-    -------
-    _replace
-        Create a new instance with changed attribute values.
-
     """
 
     za: NDArray[np.float64]
     wa: NDArray[np.float64]
-    zeff: float = 0
+    zeff: float | None = dataclasses.field(default=None)
```

Comment:
Hmm, why have it `None` as default when that `None` is never used. Having the `_calculate_zeff` value as default will also get rid of a few `type: ignore` comments. This is maybe a question for @ntessore.

Reply:
Unless the `__post_init__` function is lazy and evaluates `_calculate_zeff` only when required.

Reply:
I would set the default to `math.nan` so that we indeed never move out of the `float` space. In `__post_init__`, we can recompute `zeff` if it's `nan`. What do you think?

Reply:
Yep I think this is much better